pkg/cli/admin/release/extract_tools.go

```diff
@@ -1286,3 +1287,32 @@ func newIncluder(config manifestInclusionConfiguration) includer {
 		return m.Include(config.ExcludeIdentifier, config.RequiredFeatureSet, config.Profile, config.Capabilities, config.Overrides)
 	}
 }
+
+// ManifestReceiver has a TarEntryCallback function which can be used to as a callback to ExtractOptions.TarEntryCallback.
+// It feeds the downstream manifestsCallback
+// * with the manifests from every file whose name is not in skipNames, OR
+// * with the reader that contains the content of each file whose name is in skipNames.
+// All the errors encountered when parsing the manifests are collected in ManifestErrs.
+type ManifestReceiver struct {
+	manifestsCallback func(filename string, manifests []manifest.Manifest, reader io.Reader) (cont bool, err error)
```

Comment:
Maybe just call it `downstreamCallback`? Even your Godoc talks about it that way. Calling it `manifestCallback` is a bit confusing to me because it kinda hints that it would be called with just manifests (esp. together with the other field called 'skipNames`)